packages/connect/src/connect.js

```diff
@@ -185,13 +185,14 @@ export function connect(Comp, options) {
 }
 
 export const useConnect = () => {
-  if (!isConnected) {
-    warn(
-      "Warning: useConnect() is being used in a non connected component, " +
-        "therefore the component won't update on state changes. " +
-        "Please wrap your component with connect().\n"
-    );
-  }
+  if (!isConnected)
+    useEffect(() => {
```

Comment:
@luisherranz I'm not sure why is needed `useEffect` here. Is it a standard practice to use `useEffect` around any kind of side effect?
I guess that if we read `isConnected` inside the `useEffect`, it's value might have changed already, though implementing it this way goes against the `hook` rules and might create bugs in the future when using this hook, I think: https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

Reply:
I did it to avoid showing the warning multiple times for the same component.

But you're that a

I guess we could use an internal state to store if `isConnected` was there and then look at it inside the `useEffect`, but it seems overcomplicated just to avoid multiple warnings, which is not a big deal.

I'm going to revert it. Good catch.

Reply:
> because the if result won't change over the whole execution of the app, but you're right that it's safer to avoid it.

True. Yeah, I didn't think about that.

Reply:
@luisherranz @orballo what about using `useLayoutEffect` instead? It runs synchronously.

Reply:
@DAreRodz I just tried that, but it doesn't work :/

Reply:
:man_shrugging:

Reply:
Yeah, don't worry about this. It's not a bid deal.